File: java/byps/src-ser-bin/com/wilutions/byps/BSerializer_22.java
package com.wilutions.byps;



public class BSerializer_22 extends BSerializer {

	public final static BSerializer instance = new BSerializer_22();

	public BSerializer_22() {
		super(BRegistry.TYPEID_VALUECLASS);
	}

	public BSerializer_22(int typeId) {
		super(typeId);
	}
	
	@Override
	public int size(final Object obj, final BBinaryModel bmodel) throws BException {
		return 0;
	}

	@Override
	public void write(final Object obj1, final BOutput bout1, final int version) throws BException {
		final BOutputBin bout = ((BOutputBin)bout1);
		final BValueClass obj = (BValueClass)obj1;
		bout.bbuf.putLong(obj.changedMembers);
	}

	@Override
	public Object read(final Object obj1, final BInput bin1, final int version) throws BException {
		final BInputBin bin = ((BInputBin)bin1);
		final long v = bin.bbuf.getLong();
		BValueClass obj = (BValueClass)obj1;
		if (obj == null) {
			obj = new BValueClass();
			bin.onObjectCreated(obj);
		}
		obj.setChangedMember(v);
		return obj;
	}

}

File: csharp/byps/BSerializer_22.cs
/* USE THIS FILE ACCORDING TO THE COPYRIGHT RULES IN LICENSE.TXT WHICH IS PART OF THE SOURCE CODE PACKAGE */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace byps
{
    public class BSerializer_22 : BSerializer
    {
	    public static BSerializer instance = new BSerializer_22();

	    public BSerializer_22() 
            : base(BRegistry.TYPEID_VALUECLASS)
        {
	    }

	    public BSerializer_22(int typeId) 
            : base(typeId)
        {
	    }

        public override void write(Object obj1, BOutput bout1, long version)
        {
		    BOutputBin bout = ((BOutputBin)bout1);
		    BValueClass obj = (BValueClass)obj1;
            bout.bbuf.putLong(obj.changedMembers);
	    }

        public override Object read(Object obj1, BInput bin1, long version)
        {
		    BInputBin bin = ((BInputBin)bin1);
		    long v = bin.bbuf.getLong();
            BValueClass obj = (BValueClass)(obj1 != null ? obj1 : bin.onObjectCreated(new BValueClass()));
            obj.changedMembers = v;
		    return obj;
	    }


    }
}
